quinn-proto/src/connection/mod.rs

```diff
@@ -77,6 +77,23 @@ where
     /// Exactly one prior to `self.rem_cids.offset` except during processing of certain
     /// NEW_CONNECTION_ID frames.
     rem_cid_seq: u64,
+    rem_cid_resettoken: Option<ResetToken>,
+
+    /// Cid rotation
+    ///
+    /// Expiration timestamp of local connection IDs
+    cids_timeline: VecDeque<CidTimeStamp>,
+    /// The lifetime of Local connection IDs
+    cids_timeout: Option<Duration>,
+    /// Number of local connection IDs that have been issued in NEW_CONNECTION_ID frames.
+    cids_issued: u64,
+    /// The sequence number of active (not yet rotated/retired by peer) local connection IDs
+    cids_active_seq: HashSet<u64>,
+    /// Sequence number so far has been acked in RETIRE_CONNECTION_ID frame
+    ack_retire_prior_to: u64,
```

Comment:
It doesn't look like this is updated based on ACKs at all, but rather based on what the peer has retired. Maybe rename and reword comment?

Reply:
Yes you are right. It is the sequence number prior to which all CIDs are retired and acknowledged.